core/domain/collection_domain.py

```diff
@@ -598,6 +751,25 @@ def delete_node(self, exploration_id):
                 exploration_id)
         del self.nodes[node_index]
 
+    def add_skill(self, skill_name):
+        """Adds the new skill domain object with the specified name."""
+
+        for _, skill in self.skills.iteritems():
+            if skill.name == skill_name:
+                raise ValueError(
+                    'Skill with name "%s" already exists.' % skill_name)
+
+        skill_id = 's' + str(self.skill_id_count)
+        self.skills[skill_id] = CollectionSkill(skill_id, skill_name, [])
+        self.skill_id_count += 1
+
+    def delete_skill(self, skill_id):
+        """Deletes skill with specified id."""
+        if skill_id not in self.skills:
+            raise ValueError(
+                'Skill with ID "%s" does not exist.' % skill_id)
+        del self.skills[skill_id]
```

Comment:
Do we also need to update ``prerequisite_skills`` and ``acquired_skills`` or is that checked as part of the validation?